cvpy/tests/test_credentials.py

```diff
@@ -0,0 +1,73 @@
+import sys
+import unittest
+from pathlib import PosixPath
+
+import xmlrunner
+
+from cvpy.annotation.base.Credentials import Credentials
+
+
+class TestCredentials(unittest.TestCase):
+
+    def test_credentials_user_password(self):
+        user = 'test_user'
+        password = 'test_password'
+
+        credentials = Credentials(user=user, password=password)
+
+        self.assertEqual(credentials.user, user)
+        self.assertEqual(credentials.password, password)
+        self.assertIsNone(credentials.token)
+
+    def test_credentials_with_token(self):
+        token = 'abc1def2ghi'
+        credentials = Credentials(token=token)
+
+        self.assertIsNotNone(credentials.token)
+        self.assertEqual(credentials.get_auth_header(), dict(Authorization=f'token {token}'))
+
+    def test_credentials_token_not_set(self):
+        try:
+            user = 'test_user'
+            password = 'test_password'
+
+            credentials = Credentials(user=user, password=password)
+            credentials.get_auth_header()
+        except Exception as e:
+            self.assertEqual(str(e), 'Token is not set.')
+
+    def test_credentials_default_authfile(self):
+        default_auth_file_path = PosixPath('~/.annotation_auth').expanduser()
+        self.assertTrue(default_auth_file_path.is_file(), 'The default file ~/.annotation_auth does not exist.')
+
+        credentials = Credentials()
+
+        self.assertTrue(credentials.token or (credentials.user and credentials.password))
+
+    def test_credentials_authfile_with_token(self):
+        auth_file_path = PosixPath(TestCredentials.auth_file_with_token)
+        self.assertTrue(auth_file_path.is_file(),
+                        f'The file {TestCredentials.auth_file_with_token} does not exist.')
+
+        credentials = Credentials(auth_file=TestCredentials.auth_file_with_token)
+        self.assertIsNotNone(credentials.token)
+
+    def test_credentials_authfile_with_user_password(self):
+        auth_file_path = PosixPath(TestCredentials.auth_file_with_user_password)
+        self.assertTrue(auth_file_path.is_file(),
+                        f'The file {TestCredentials.auth_file_with_user_password} does not exist.')
+
+        credentials = Credentials(auth_file=TestCredentials.auth_file_with_user_password)
+        self.assertIsNotNone(credentials.user)
+        self.assertIsNotNone(credentials.password)
+
+
+if __name__ == '__main__':
+    if len(sys.argv) > 1:
+        TestCredentials.auth_file_with_user_password = sys.argv.pop()
+        TestCredentials.auth_file_with_token = sys.argv.pop()
+
+    unittest.main(
+        testRunner=xmlrunner.XMLTestRunner(output='test-reports'),
+        failfast=False, buffer=False, catchbreak=False
+    )
```

Comment:
This is the first time I'm reviewing CVPy change. Will these tests be run automatically in future pushes? Or do we need to  do something else to make that happen, like modifying a Jenkins file? Not that we're about to add a lot to CVPy, we should make sure existing tests are being run as code is updated.

Reply:
I will check with @courtneyambrozic.

Reply:
Yes, we'll have to modify the Jenkinsfile to run these new tests: https://gitlab.sas.com/computer-vision/devops/jenkins/cvpy-pipeline/-/blob/master/Jenkinsfile

Mu is adding a new test file too, so maybe we can create a story to modify the Jenkinsfile to run both of these.

The Jenkins pipeline isn't set up to run for every new pull request (this is something that we discussed early on but posed a potential security risk). Currently, we just run the Jenkins job manually using the PR number as an input and then again when the PR is merged.

Reply:
We should not need to modify Jenkins file as we add more classes and unit tests in the future. I created a ticket (COMPVIS-4123) to address this issue.